packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -499,12 +499,24 @@ class DevFS {
 
     // Update modified files
     final Map<Uri, DevFSContent> dirtyEntries = <Uri, DevFSContent>{};
-
     int syncedBytes = 0;
+    if (fullRestart) {
+      generator.reset();
+    }
+    // On a full restart, or on an initial compile for the attach based workflow,
```

Comment:
This logic is unchanged, but the asset stat was placed in between the `pendingCompilerOutput ` resolution.